MonkeyWrench.Web.UI/EditLane.aspx.cs

```diff
@@ -705,7 +706,16 @@ protected void cmdSave_Click (object sender, EventArgs e)
 		lane.enabled = chkEnabled.Checked;
 		lane.additional_roles = txtRoles.Text;
 		lane.priority = Int32.Parse (lstPriority.SelectedItem.Value);
-		lane.is_protected = chkProtected.Checked;
+		lane.is_protected = chkProtected.Checked;
+
+		// Don't ever pull more than 100 commits
+		try {
+			int commits = Int32.Parse (txtFetchCommits.Text);
+			lane.max_commits_to_fetch = commits >= 100 ? 100 : commits;

```

Comment:
There is no hard limit on the database, so it would be feasible to do an update through SQL. 

A case where this could affect someone is when someone creates a new lane setting `max_commits_to_fetch` to some terrible number (>9000) around the same time someone else pushes a single commit to a branch. This person waiting for a single build now has to wait for the other 9000 builds (of which we probably only realistically care about 1 build) to be created before they get to assigned to any bot.

I can remove the limit if you like. I just don't think there is any practicality in keeping it limitless -- jenkins and vsts only build the topmost commit.